lib/song.rb

```diff
@@ -0,0 +1,25 @@
+require_relative "verses"
+
+class Song
+
+  def initialize(verses)
+    @verses = verses
+  end
+
+  def create_song(number_of_bottles)
+    #verses = []
+    until end_of_song?(number_of_bottles) do
+      @verses.verse_type(number_of_bottles)
+      number_of_bottles -= 1
+    end
+    #verses
+  end
+
+  def complete_song
```

Comment:
what is this empty method for?  Were you thinking to put the method with this name from Display here?
You may be on to something as I don't think 'run_song' method in Display is the right place.  You need a class which pulls all of the 'pieces' together, eg. collating input and then creating the song.